### PR TITLE
Extract `incrementalmerkletree::testing` module to a separate `incrementalmerkletree-testing` crate.

### PR DESCRIPTION
**Pull request by nuttycom**
This change will allow us to continue to move `incrementalmerkletree` and `shardtree` development forward without having to continue to maintain `bridgetree` compatibility with `incrementalmerkletree::testing` after the next `incrementalmerkletree` release.

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.60.0"
+channel = "1.70.0"
```
Comment (str4d) [inline rust-toolchain.toml:2]: This changes alters the Rust version that CI checks MSRV against. Per https://github.com/zcash/librustzcash/issues/759 we need at least 1.64 use workspaces, but I don't recall if using workspaces prevents the crate from being depended on downstream with a lower MSRV. How is the `Cargo.toml` adjusted during publishing for workspace dependencies?

File: shardtree/src/lib.rs [hunk 2]
```diff
@@ -1462,9 +1461,7 @@ mod tests {
             // boundary.
             tree.batch_insert(
                 frontier_end + 1,
-                ('g'..='j')
-                    .into_iter()
-                    .map(|c| (c.to_string(), Retention::Ephemeral)),
+                ('g'..='j').map(|c| (c.to_string(), Retention::Ephemeral)),
```
Comment (str4d) [inline shardtree/src/lib.rs:1464]: I think these changes in the second commit are silently adapting to the silent MSRV bump done in the first commit, specifically to clippy lint changes. I do not know whether these changes are actually compatible with an MSRV of 1.60; this needs testing.

**Review by str4d — COMMENTED** [review]
Partial review as of d3db03fe67066df02cb727bc465743c72395c0bf (I'm still checking the large move).

If it turns out that the workspace change does require downstreams to raise MSRV, I would prefer that we explicitly raise MSRV in its own commit instead of implicitly doing.

I'd mildly perfer 1.64 instead of 1.70 for this PR (as that is the only necessary bump; it could be raised again in a subsequent PR as new features are needed).

File: rust-toolchain.toml (hunk 1 above)
Comment (nuttycom) [reply rust-toolchain.toml:2]: I'll try 1.64

**Review by str4d — COMMENTED** [review]
Reviewed d3db03fe67066df02cb727bc465743c72395c0bf

File: rust-toolchain.toml (hunk 1 above)
Comment (nuttycom) [reply rust-toolchain.toml:2]: From what I can tell, I can't build even the individual crates without 1.64 when using common workspace stanzas. I think that 1.64 is an acceptable lower bound on MSRV though.

**Review by str4d — APPROVED** [review]
utACK f6a5ae46e36a55395b44f7372698b65f916b72b5

**Review by str4d — APPROVED** [review]
Re-utACK d663a8f64f04832ebb3b9dfac868ff3ae7b805de

**Pull request merged**